### PR TITLE
Add Service Worker to enable offline Viewer

### PR DESCRIPTION
**Pull request by andrecardoso**
## Description
- Add 'generate-service-worker' gulp task to build script
- service-worker.js file is generated at src/main/java/webapp/ to be bundled and served along with the other files
- Viewer can run HTML Templates offline

## Motivation and Context
Part of the Simplify PUD epic, this is to enable HTML Templates running on Viewer while offline

## How Has This Been Tested?
Tested by loading `https://viewer-test.risevision.com` on the player and launching it while offline

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

File: src/main/webapp/service-worker.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
+/* eslint-disable quote-props */
```
Comment (andrecardoso) [inline src/main/webapp/service-worker.js]: @alex-deaconu @olegrise I'm not so sure we should add this to git as it is going to be generated on every build. On the other hand, it is clear to everybody what are the current precached versions for each file. What is your opinion? Thanks!

**Comment by andrecardoso** [comment]
@alex-deaconu could you please review this?  Thanks!

**Comment by andrecardoso** [comment]
[(3) Add service workers to Viewer](https://trello.com/c/awpYIArn/365-3-add-service-workers-to-viewer)

File: src/main/webapp/service-worker.js (hunk 1 above)
Comment (alex-deaconu) [reply src/main/webapp/service-worker.js]: @andrecardoso my preference is not to commit. This file could easily get out of date if a developer forgets to run the build before committing their changes. If we want to see the versions, we can just access the file directly for the specific version in App Engine. Thoughts? Thanks.

**Comment by alex-deaconu** [comment]
@andrecardoso reviewed the code and did a quick online test and I didn't find any issues. However, I did not validate on Player; I'm assuming you've done all that testing yourself. Looks good to me 👍. Thanks!

File: src/main/webapp/service-worker.js (hunk 1 above)
Comment (olegrise) [reply src/main/webapp/service-worker.js]: @andrecardoso I tend to agree with @alex-deaconu.

Comment (andrecardoso) [reply src/main/webapp/service-worker.js]: OK, I'm going to remove it. Thanks!

**Pull request merged**